### DISCUSSION TITLE
Why does my "spaces" query return bsp on type when it is stacked?

### DISCUSSION THREAD

**Discussion by mastoj** [asmvik/yabai#1795] [Q&A] [answered]
I was planning to create a simple script that would either:

1. Move a window on top of another one and create a stack
2. If it is already in a stack it would move it out.

To create such a script I need to be able to check the type of a space, but it always returns "bsp". What do I need to do for a stack to return correct for `yabai -m query --spaces --space`? 

The output I get is:

```
{
        "id":12,
        "uuid":"DD832499-2F11-47BB-B93E-748F2CE0392B",
        "index":4,
        "label":"",
        "type":"bsp",
        "display":2,
        "windows":[159, 94, 76],
        "first-window":159,
        "last-window":159,
        "has-focus":true,
        "is-visible":true,
        "is-native-fullscreen":false
}
```

I expected `type` to be "stack" in my case.

**Comment by asmvik** [comment, marked as answer]
Because the space itself is using a bsp layout. The two windows are stacked, meaning they share the same node in the bsp-layout.
You want to query information about the window (`yabai -m query --windows --window` for the focused window, use a selector to get a different window), and check the `stack-index` property. If non-zero, the window is inside a stack, and if zero it is not stacked.

**Reply by mastoj** [reply]
Thanks! Exactly what I needed!
